Quick edit modal for content language

AlexVelezLl opened the pull request:
## Summary
### Description of the change(s) you made

Add a quick edit modal to edit content nodes languages. Also add an checkbox to update descendants languages, but this does not work for now until #4372 is merged.

### Screenshots

https://github.com/learningequality/studio/assets/51239030/d78536f3-373f-48da-b939-aac0c0b32388
___
## Reviewer guidance
### How can a reviewer test these changes?

1. Go to update a channel content
2. Click on the "Edit Language" option on the content node menu.
3. Or you can select multiple content nodes and select "Edit language" on the bulk toolbar.

## References
Closes #3415

## Comments
The "clear" button on the search textbox should be included on KDS first, and we should add just a prop in this code.

----

### Contributor's Checklist
<!-- After saving the PR, come through to tick off completed checklist items. Delete any sections that are not applicable to your PR -->

Studio-specifc:

- [ ] All user-facing strings are translated properly
- [ ] The `notranslate` class been added to elements that shouldn't be translated by Google Chrome's automatic translation feature (e.g. icons, user-generated text)
- [ ] All UI components are LTR and RTL compliant
- [ ] Views are organized into `pages`, `components`, and `layouts` directories [as described in the docs](https://github.com/learningequality/studio/blob/vue-refactor/docs/architecture.md#where-does-the-frontend-code-live)
- [ ] Users' storage used is recalculated properly on any changes to main tree files
- [ ] If there new ways this uses user data that needs to be factored into our [Privacy Policy](https://github.com/learningequality/studio/tree/master/contentcuration/contentcuration/templates/policies/text), it has been noted.


Testing:

- [x] Code is clean and well-commented
- [x] Contributor has fully tested the PR manually
- [x] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Any new interactions have been added to the [QA Sheet](https://docs.google.com/spreadsheets/d/1HF4Gy6rb_BLbZoNkZEWZonKFBqPyVEiQq4Ve6XgIYmQ/edit#gid=0)
- [x] Critical and brittle code paths are covered by unit tests
___

### Reviewer's Checklist
#### This section is for reviewers to fill out.

- [ ] Automated test coverage is satisfactory
- [ ] PR is fully functional
- [ ] PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- [ ] External dependency files were updated if necessary (`yarn` and `pip`)
- [ ] Documentation is updated
- [ ] Contributor is in AUTHORS.md

akolson commented on contentcuration/contentcuration/frontend/channelEdit/components/QuickEditModal/__tests__/EditTitleDescriptionModal.spec.js:
Using setTimeout in tests is not recommended may make the tests unreliable. You could consider using `requestAnimationFrame` instead.

akolson replied:
Here is an [example](https://github.com/Soren568/kolibri-design-system/blob/d7a0c1e232bbe96667c57afafd762f3342fdf491/lib/__tests__/useKWindowDimensions.spec.js#L45C24-L45C45) that uses the `requestAnimationFrame `

akolson commented on contentcuration/contentcuration/frontend/channelEdit/components/QuickEditModal/__tests__/EditLanguageModal.spec.js:
Please see https://github.com/learningequality/studio/pull/4355/files#r1444647223 for consideration!

akolson commented on contentcuration/contentcuration/frontend/channelEdit/components/QuickEditModal/EditLanguageModal.vue:
```diff
@@ -0,0 +1,182 @@
+        }
+        const criteria = ['id', 'native_name', 'readable_name'];
+        return LanguagesList.filter(lang =>
+          criteria.some(key => lang[key]?.toLowerCase().includes(searchQuery))
```
It's worth-noting that the chaining operator(`?.`) is [not supported in IE 11](https://caniuse.com/?search=%3F.) that we are still supporting(The last time i checked) :(.

akolson reviewed (commented):
Implementation looks correct to me! I have left a few general comments for your consideration. Great work @AlexVelezLl

AlexVelezLl commented:
Thank! @akolson I have replaced the setTimeout (didnt know about the requestAnimationFrame, thank you!!), and removed the chaning operator :raised_hands:.
